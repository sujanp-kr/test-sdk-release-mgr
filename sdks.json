{
  "banner": {
    "sshURL": "git@github.com:krogertechnology/behavioral-analytics-banner.git",
    "htmlURL": "https://github.com/krogertechnology/behavioral-analytics-banner",
    "repo": "behavioral-analytics-banner",
    "weeklyUpdates": {
      "outputDest": "behavioral-analytics-banner/src",
      "schemaRoot": "banner"
    },
    "consumer": {}
  },
  "demeter": {
    "sshURL": "git@github.com:krogertechnology/behavioral-analytics-demeter.git",
    "htmlURL": "https://github.com/krogertechnology/behavioral-analytics-demeter",
    "repo": "behavioral-analytics-demeter",
    "weeklyUpdates": {
      "outputDest": "behavioral-analytics-demeter/src",
      "schemaRoot": "demeter"
    },
    "consumer": {}
  },
  "devmarketplace": {
    "sshURL": "git@github.com:krogertechnology/behavioral-analytics-devmarketplace.git",
    "htmlURL": "https://github.com/krogertechnology/behavioral-analytics-devmarketplace",
    "repo": "behavioral-analytics-devmarketplace",
    "weeklyUpdates": {
      "outputDest": "behavioral-analytics-devmarketplace/src",
      "schemaRoot": "devmarketplace"
    },
    "consumer": {}
  },
  "esperanto": {
    "sshURL": "git@github.com:krogertechnology/esperanto.git",
    "htmlURL": "https://github.com/krogertechnology/esperanto",
    "repo": "esperanto",
    "weeklyUpdates": {
      "outputDest": "esperanto/@kroger/clickstream/analytics-utils/types",
      "schemaRoot": "banner"
    },
    "consumer": {}
  },
  "usageMetrics": {
    "sshURL": "git@gitlab.kroger.com:kat/web/analytics-sdk.git",
    "htmlURL": "https://gitlab.kroger.com/kat/web/analytics-sdk/",
    "repo": "Not github repo",
    "weeklyUpdates": {},
    "consumer": ["analytics-core", "esperanto"]
  },
  "analytics-core": {
    "sshURL": "git@github.com:krogertechnology/behavioral-analytics-core.git",
    "htmlURL": "https://github.com/krogertechnology/behavioral-analytics-core",
    "repo": "behavioral-analytics-core",
    "weeklyUpdates": {},
    "consumer": {
      "include": [
        {"orgRepo": "krogertechnology/behavioral-analytics-banner"},
        {"orgRepo": "krogertechnology/behavioral-analytics-demeter"},
        {"orgRepo": "krogertechnology/behavioral-analytics-devmarketplace"}
      ]
    }
  },
  "lodash": {
    "sshURL": "",
    "htmlURL": "",
    "repo": "Not KAT repo; Only for testing",
    "weeklyUpdates": {},
    "consumer": {
          "includes": [
            {"orgRepo": "sujanp-kr/consumer-A"},
            {"orgRepo": "sujanp-kr/consumer-B"}
          ]
    }
  }
}
