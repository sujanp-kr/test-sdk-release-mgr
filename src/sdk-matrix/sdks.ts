import fs from "fs";

const FILE_NAME = "sdks.json";

export type WeeklyUpdateData = {
  outputDest: string;
  schemaRoot: string;
};

export type KatSDK = {
  sshURL: string;
  htmlURL: string;
  repo: string;
  weeklyUpdates: WeeklyUpdateData;
  // consumers: string[];
};

export type KatSDKMap = {
  [name: string]: KatSDK;
};

export type Consumer = {
  sdkName: string;
  consumer: string[];
};

export type ConsumerMap = {
  [name: string]: Consumer;
};

export function getKatSdks(fileName = FILE_NAME): KatSDKMap {
  const json: KatSDKMap = JSON.parse(fs.readFileSync(fileName).toString());
  return json;
}

export function getConsumers(): ConsumerMap {
  const json: ConsumerMap = JSON.parse(
    fs.readFileSync("consumers.json").toString()
  );
  return json;
}
