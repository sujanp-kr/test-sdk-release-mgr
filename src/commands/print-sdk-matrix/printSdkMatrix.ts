import yargs from "yargs";
import { KatSDK, KatSDKMap, getKatSdks } from "../../sdk-matrix/sdks";

interface GET_SDK_MATRIX extends KatSDK {
  name: string;
}

export function getSdkMatrix(
  sdkMap: KatSDKMap = getKatSdks()
): GET_SDK_MATRIX[] {
  return Object.entries(sdkMap).map(([name, details]) => ({
    name,
    ...details,
  }));
}

/**
 * Used to provide a matrix to pass back to Github actions
 *
 * https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix
 */
export function printSdkMatrix(): void {
  const sdkMatrix = getSdkMatrix();
  const sdkOptions = sdkMatrix.map(({ name }) => name);

  const { sdk } = yargs(process.argv)
    .options({
      sdk: {
        demandOption: false,
        choices: [...sdkOptions, "all"],
      },
    })
    .parseSync();
  console.log(
    JSON.stringify(
      sdk !== "all" ? sdkMatrix.filter(({ name }) => name === sdk) : sdkMatrix
    )
  );
}
