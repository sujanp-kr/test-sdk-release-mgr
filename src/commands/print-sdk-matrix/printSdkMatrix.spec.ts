import { getSdkMatrix } from "./printSdkMatrix";
import { KatSDKMap } from "../../sdk-matrix/sdks";

describe("getSdkMatrix", () => {
  it("should returns sdk names", () => {
    const mockSDK: KatSDKMap = {
      banner: {
        sshURL: "sshURL",
        htmlURL: "htmlURL",
        repo: "brepo",
        weeklyUpdates: {
          outputDest: "behavioral-analytics-banner/src",
          schemaRoot: "banner",
        },
      },
      demeter: {
        sshURL: "sshURL",
        htmlURL: "htmlURL",
        repo: "brepo",
        weeklyUpdates: {
          outputDest: "behavioral-analytics-demeter/src",
          schemaRoot: "demeter",
        },
      },
      devmarketplace: {
        sshURL: "sshURL",
        htmlURL: "htmlURL",
        repo: "brepo",
        weeklyUpdates: {
          outputDest: "behavioral-analytics-devmarketplace/src",
          schemaRoot: "devmarketplace",
        },
      },
    };

    expect(getSdkMatrix(mockSDK)).toMatchSnapshot();
  });
});
