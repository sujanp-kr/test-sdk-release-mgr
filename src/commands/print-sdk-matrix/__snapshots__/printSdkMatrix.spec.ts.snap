// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSdkMatrix should returns sdk names 1`] = `
Array [
  Object {
    "htmlURL": "htmlURL",
    "name": "banner",
    "repo": "brepo",
    "sshURL": "sshURL",
    "weeklyUpdates": Object {
      "outputDest": "behavioral-analytics-banner/src",
      "schemaRoot": "banner",
    },
  },
  Object {
    "htmlURL": "htmlURL",
    "name": "demeter",
    "repo": "brepo",
    "sshURL": "sshURL",
    "weeklyUpdates": Object {
      "outputDest": "behavioral-analytics-demeter/src",
      "schemaRoot": "demeter",
    },
  },
  Object {
    "htmlURL": "htmlURL",
    "name": "devmarketplace",
    "repo": "brepo",
    "sshURL": "sshURL",
    "weeklyUpdates": Object {
      "outputDest": "behavioral-analytics-devmarketplace/src",
      "schemaRoot": "devmarketplace",
    },
  },
]
`;
